anchore_engine/services/catalog/__init__.py

```diff
@@ -402,6 +402,18 @@ def handle_vulnerability_scan(*args, **kwargs):
     return True
 
 
+def generate_error_service_description(error_short_description: str) -> str:
```

Comment:
Do you mean an exception?  There isn't always an exception thrown when this is invoked. The issue that I'm trying to solve is that the service_detail in the service record gets populated with strings instead of something shaped like service_statuses in anchore_engine/subsys/servicestatus.py line 60. I think it would be ideal if there were objects that modeled that structure (instead of a dictionary), but that change would require a nontrivial rework of the existing code. If I totally misunderstood the question, please let me know.

Reply:
@Vijay-P I meant what you said around objects to model the structure instead of a dict. But if it's non-trivial maybe we can open up a follow up improvement task?

Reply:
+1 for follow up improvement